src/utils.jl

```diff
@@ -21,12 +21,27 @@ macro check_args(K, param, cond, desc=string(cond))
     end
 end
 
-function vec_of_vecs(X::AbstractMatrix; obsdim::Int=2)
-    @assert obsdim ∈ (1, 2) "obsdim should be 1 or 2, see docs of kernelmatrix"
-    if obsdim == 1
-        RowVecs(X)
+function deprecated_obsdim(obsdim::Union{Int,Nothing})
+    return if obsdim === nothing
+        Base.depwarn(
+            "implicit `obsdim=2` argument is deprecated and now has to be passed " *
+            "explicitly to specify that each column corresponds to one observation",
+            :vec_of_vecs,
+        )
+        2
     else
+        obsdim
+    end
+end
+
+function vec_of_vecs(X::AbstractMatrix; obsdim::Union{Int,Nothing}=nothing)
+    _obsdim = deprecated_obsdim(obsdim)
+    return if _obsdim == 1
+        RowVecs(X)
+    elseif _obsdim == 2
         ColVecs(X)
```

Comment:
Likewise, I find
```suggestion
    if _obsdim == 1
        return RowVecs(X)
    elseif _obsdim == 2
        return ColVecs(X)
```
much easier to parse

Reply:
Same here, I prefer the version in this PR.